apps/test/solutions/calc/displayComplexUserExpressions.js

```diff
@@ -205,6 +205,131 @@ function customValidator(assert) {
     validateTextElement(g.children[2], '204', null);
   });
 
+  displayComplexUserExpressionTest(assert, 'divide by zero error', function () {
+    // compute: f(10)
+    // f(i) = (4 / (4 - 4))
+    var userSet = new EquationSet();
+    userSet.addEquation_(new Equation(null, [], new ExpressionNode('f', [10])));
+    userSet.addEquation_(new Equation('f', ['i'], new ExpressionNode('/', [
+      new ExpressionNode(4),
+      new ExpressionNode('-', [4, 4])
+    ])));
+
+    // target is similar, but no div 0
+    // compute: f(10)
+    // f(i) = (4 / (5 - 4))
+    var targetSet = new EquationSet();
+    targetSet.addEquation_(new Equation(null, [], new ExpressionNode('f', [10])));
+    targetSet.addEquation_(new Equation('f', ['i'], new ExpressionNode('/', [
+      new ExpressionNode(4),
+      new ExpressionNode('-', [4, 4])
+    ])));
+
+    setEquationSets(targetSet, userSet);
+
+    displayComplexUserExpressions();
+
+    assert.equal(userExpression.children.length, 2);
+
+    // line 1: f(i) = (4 / (4 - 4))
+    var g = userExpression.children[0];
+    validateTextElement(g.children[0], 'f(i) = ', null);
```

Comment:
Seems like a helper could maybe be handy here, `validateChildrenMatch(g, ['(', '4', ' / '` [...]